fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -137,19 +142,23 @@
     private int limit;  // Not final as may be adjusted when running.
     private final int maxRetries;
     private final int recordsPerSecond;
+    private final long progressLogIntervalMillis;
+    private long timeOfLastProgressLogMillis;
 
     protected OnlineIndexer(@Nonnull FDBDatabaseRunner runner,
                             @Nonnull FDBRecordStore.Builder recordStoreBuilder,
                             @Nonnull Index index, @Nonnull Collection<RecordType> recordTypes,
-                            int limit, int maxRetries, int recordsPerSecond) {
+                            int limit, int maxRetries, int recordsPerSecond, long progressLogIntervalMillis) {
         this.runner = runner;
         this.recordStoreBuilder = recordStoreBuilder;
         this.index = index;
         this.recordTypes = recordTypes;
         this.limit = limit;
         this.maxRetries = maxRetries;
         this.recordsPerSecond = recordsPerSecond;
+        this.progressLogIntervalMillis = progressLogIntervalMillis;
         this.recordsRange = computeRecordsRange();
+        timeOfLastProgressLogMillis = System.currentTimeMillis();
```

Comment:
I suppose it's intentional that the first progress statement won't happen until `progressLogIntervalMillis` milliseconds after object creation? I'm not sure what the alternative would be other than to have it log immediately (by setting the time of last progress millis to, like, `Integer.MIN_VALUE` or 0 if we're okay with this only working after Jan 1, 1970).

Reply:
Yes, it was intentional to not have it log until the interval has passed, mostly so that it doesn't log anything if the index builds really fast, just the other logging that occurs before/after index building.